pkg/operator/controller/ingress/deployment.go

```diff
@@ -560,6 +563,39 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 	}
 	env = append(env, corev1.EnvVar{Name: RouterHAProxyThreadsEnvName, Value: strconv.Itoa(threads)})
 
+	var responseHeaderValues string
+	var responseHeaderSpecs []string
+	var requestHeaderValues string
+	var requestHeaderSpecs []string
+
+	if ci.Spec.HTTPHeaders != nil && ci.Spec.HTTPHeaders.Actions.Response != nil && len(ci.Spec.HTTPHeaders.Actions.Response) != 0 {
+		for _, value := range ci.Spec.HTTPHeaders.Actions.Response {
+			if value.Action.Type == operatorv1.Set && value.Action.Set != nil && len(value.Name) != 0 {
+				responseHeaderSpecs = append(responseHeaderSpecs, getHTTPHeadersListForSetOrReplaceOrAppend(value))
+				responseHeaderValues = strings.Join(responseHeaderSpecs, ",")
+			}
+			if value.Action.Type == operatorv1.Delete && len(value.Name) != 0 {
+				responseHeaderSpecs = append(responseHeaderSpecs, getHTTPHeadersForDelete(value))
+				responseHeaderValues = strings.Join(responseHeaderSpecs, ",")
+			}
+		}
+		env = append(env, corev1.EnvVar{Name: "ROUTER_HTTP_RESPONSE_HEADERS", Value: responseHeaderValues})
+	}
+
+	if ci.Spec.HTTPHeaders != nil && ci.Spec.HTTPHeaders.Actions.Request != nil && len(ci.Spec.HTTPHeaders.Actions.Request) != 0 {
```

Comment:
```suggestion
	if ci.Spec.HTTPHeaders != nil && ci.Spec.HTTPHeaders.Actions != nil && len(ci.Spec.HTTPHeaders.Actions.Request) != 0 {
```

Reply:
ci.Spec.HTTPHeaders.Actions does not work...it is not a pointer.

Reply:
May be I need to explain this basic concept as we are going in a loop for this discussion. Using a pointer type reduces visibility.  Using a non-pointer type means that the API server always includes the field when it generates json/yaml for an object.

Reply:
```suggestion
	if ci.Spec.HTTPHeaders != nil && ci.Spec.HTTPHeaders.Actions.Request != "" {
```

Reply:
@candita request is an array so we shall check the length.

Reply:
Then you can use `len` on an empty array and don't need to check nil.
```suggestion
	if ci.Spec.HTTPHeaders != nil && len(ci.Spec.HTTPHeaders.Actions.Request) != 0 {
```